src/MessagePack/Resolvers/DynamicObjectResolver.cs

```diff
@@ -1750,9 +1750,17 @@ bool AddEmittableMemberOrIgnore(bool isIntKeyMode, EmittableMember member, bool
                     if (isIntKeyMode ? intMembers.TryGetValue(member.IntKey, out var conflictingMember) : stringMembers.TryGetValue(member.StringKey!, out conflictingMember))
                     {
                         // Quietly skip duplicate if this is an override property.
-                        if (member.PropertyInfo != null && ((conflictingMember.PropertyInfo?.SetMethod?.IsVirtual ?? false) || (conflictingMember.PropertyInfo?.GetMethod?.IsVirtual ?? false)))
+                        if (member.PropertyInfo is not null && conflictingMember.PropertyInfo is not null
+                            && member.PropertyInfo.Name == conflictingMember.PropertyInfo.Name)
                         {
-                            return false;
+                            // Implicit interface implementations of get/set methods are marked 'virtual final'
+                            // Therefore an overridable property should be 'IsVirtual == true && IsFinal == false'
```

Comment:
Not necessarily. Ordinary properties (without interfaces) [can also be virtual and final](https://sharplab.io/#v2:C4LglgNgPgAgTAAgIIIN4FgBQCcJgZgQDcwAnYAVwEMIEAjAewdoAVSGAHNBAcwFNgAbgQBnAcIC+WKZizwEAIQQhkaLLjyExNPgBMEDIn1KkwuvvSat2XVL3GiHMiUA).

Reply:
How about this:
```cs
// According to MethodBase.IsVirtual docs an overridable property should be 'IsVirtual == true && IsFinal == false'.
// Property methods can be marked 'virtual final' in case of 'sealed override' or when implementing interface implicitly.
```